src/map/utils/battleutils.cpp

```diff
@@ -7058,4 +7059,53 @@ namespace battleutils
         return 1.0;
     }
 
+    uint8 TryProcTreasureHunter(CCharEntity* PChar, CMobEntity* PMob, CAttackRound* attackRound, bool feintApplied)
+    {
+        // The Treasure Hunter proc system was introduced in the December 2010 update and applies to players using Thief main job, as well as the Ranger job ability Bounty Shot.
+        // https://www.bg-wiki.com/ffxi/Treasure_Hunter
+        // https://ffxiclopedia.fandom.com/wiki/Treasure_Hunter
+
+        auto playerTH = PChar->getMod(Mod::TREASURE_HUNTER);
+        auto mobTH    = PMob->PEnmityContainer->GetHighestTH();
```

Comment:
I'm pretty sure the TH level of the mob is tracked independently from enmity. Wiki says player can leave the zone/party and TH will only reset when mob goes passive (it's not clear if TH resets as soon as it goes passive or until it starts to regen health). I'm not sure how the system currently works but there is m_THLevel or something on the mob entity that I assume would only need some tweaking to work with the new system.